src/main/java/com/flab/bbt/aop/LoginCheckAspect.java

```diff
@@ -0,0 +1,29 @@
+package com.flab.bbt.aop;
+
+import com.flab.bbt.common.SessionConst;
+import com.flab.bbt.exception.CustomException;
+import com.flab.bbt.exception.ErrorCode;
+import com.flab.bbt.user.domain.User;
+import javax.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.stereotype.Component;
+
+@Component
+@Aspect
+@RequiredArgsConstructor
+public class LoginCheckAspect {
+
+    private final HttpSession session;
+
+    @Before("@annotation(com.flab.bbt.aop.LoginCheck) "
+        + "&& execution(* com.flab.bbt..*Controller.*(..))")
+    public void loginCheck() {
+        User user = (User) session.getAttribute(SessionConst.COOKIE_SESSION_ID);
+
+        if (user == null) {
+            throw new CustomException(ErrorCode.USER_UNAUTHORIZED);
+        }
```

Comment:
1.
null check만 하는거라면 아래와같이 할 수 있을것 같습니다.
```suggestion
        if (session.getAttribute(SessionConst.COOKIE_SESSION_ID) == null) {
            throw new CustomException(ErrorCode.USER_UNAUTHORIZED);
        }
```

2. 
AOP동작시 `RestController`에있는 코드은 중복코드가 되니 제거해도 됩니다.
<img width="578" alt="image" src="https://user-images.githubusercontent.com/105277168/213318696-3794c0c0-601c-4e83-bfb4-3c62189d9233.png">

Reply:
@f-lab-lion 
2번 관련해서, 해당 컨트롤러에서는 로그인 검증 뿐만 아니라 세션에서 값을 가져와 직접 사용을 하고 있습니다(유저 정보가 필요해 세션을 통해 갖고 오고 있습니다). 이럴 경우에는 제거를 안해도 되는지, 아니면 중복제거를 위해 다른 방법을 생각해봐야 하는건지 궁금합니다. 중복제거해보기 위해 고민해보긴 했는데 딱히 떠오르는 아이디어가 없네요ㅜㅜ